src/vs/workbench/common/configuration.ts

```diff
@@ -221,7 +221,7 @@ export class DynamicWorkbenchSecurityConfiguration extends Disposable implements
 					},
 					'default': [],
 					'markdownDescription': localize('security.allowedUNCHosts', 'A set of UNC host names (without leading or trailing backslash, for example `192.168.0.1` or `my-server`) to allow without user confirmation. If a UNC host is being accessed that is not allowed via this setting or has not been acknowledged via user confirmation, an error will occur and the operation stopped. A restart is required when changing this setting. Find out more about this setting at https://aka.ms/vscode-windows-unc.'),
-					'scope': ConfigurationScope.MACHINE
+					'scope': ConfigurationScope.APPLICATION_MACHINE
```

Comment:
As discussed, make the other setting `security.restrictUNCAccess` also application machine.